graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py

```diff
@@ -1919,6 +1926,35 @@ def multinomial_sample(label_col: str) -> Sequence[int]:
 
         return train_mask_df, val_mask_df, test_mask_df
 
+    @staticmethod
+    def _is_multitask(data_configs: Mapping[str, Sequence[StructureConfig]]) -> bool:
+        """Returns True when the input configuration contains more than one label.
```

Comment:
A case is that if users now build a graph with both node classification label and link prediction label, I think our current code will treat them as a multi-task graph construction and build graph with new mask name like ``train_mask_lp``.

However, if user just want to create a graph with both node classification label and link prediction label, and then separately launch node classification task and link prediction task. The design here will meet the backward compatibility problem because the mask name is already changed. In this way we are forcing the user to use multi-task training framework.

Reply:
Yes I just submitted an RFC that raises that: https://github.com/awslabs/graphstorm/issues/997

I think it's an OK feature to drop, I don't know of any user that would create multiple individual labels. And it might be possible to also fix downstream, we can use `str.startswith()` when determining which features to load as masks.

Reply:
Or maybe a more appropriate example is that we create labels on multiple edges for link prediction and we do training on these multiple types of edges which can be a common case for users. Previously we construct a graph with all named by default with `train_mask`, and the training will go to the link prediction training workflow. But now it will not support this case any more. Users need to go to the multi-task flow to do all of these.

Reply:
Yeah that's a legitimate usecase.

One way we could have both is by modifying [GSgnnData.get_edge_train_set()](https://github.com/thvasilo/graphstorm/blob/ca07f7f2f020202223ae222e9c89d40a2cc7d591/python/graphstorm/dataloading/dataset.py#L764) to use the `train_mask` _prefix_ to retrieve the train masks. This I think ensures backwards compatibility.